deeppavlov/skills/ecommerce_bot/bot.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2018 Neural Networks and Deep Learning lab, MIPT
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import json
+
+from collections import Counter
+from typing import List, Tuple, Dict, Any
+from operator import itemgetter
+from scipy.stats import entropy
+
+import numpy as np
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.common.file import save_pickle, load_pickle
+from deeppavlov.core.commands.utils import expand_path, make_all_dirs, is_file_exist
+from deeppavlov.core.models.estimator import Component
+from deeppavlov.metrics.bleu import bleu_advanced
+
+log = get_logger(__name__)
+
+@register("ecommerce_bot")
+class EcommerceBot(Component):
+    """Class to retrieve product items from `load_path` catalogs
+    in sorted order according to the similarity measure
+    Retrieve the specification attributes with corresponding values
+    in sorted order according to entropy.
+
+    Parameters:
+        min_similarity: similarity threshold for ranking
+        min_entropy: min entropy threshold for specifying
+        entropy_fields: the specification attributes of the catalog items
+        preprocess: text preprocessing component
+
+    Returns:
```

Comment:
A class docstring should describe initialization parameters, and return values are better suited for __call__ method docstring.
Also, the order of parameters is better to be consistent with the method signature.